homeassistant/components/mqtt/mixins.py

```diff
@@ -276,7 +276,8 @@ def async_handle_schema_error(
 async def _async_discover(
     hass: HomeAssistant,
     domain: str,
-    async_setup: partial[Coroutine[Any, Any, None]],
+    setup: partial[CALLBACK_TYPE] | None,
+    async_setup: partial[Coroutine[Any, Any, None]] | None,
```

Comment:
Maybe a comment here that setup is to be run in the event loop when there is nothing to be awaited?

Reply:
Added a comment to the docstr